### PR TITLE
feat: add eslint and prettier

### PR DESCRIPTION
**Pull request by domharrington**
## 🧰 Changes

Add eslint and prettier and fix style issues
Add test for single quoted yaml files (which broke when i prettier'd the yaml fixture)

## 🧬 QA & Testing

Do the tests pass?

### CODE REVIEW

**Comment by domharrington** [comment]
The tests aren't running yet, because this PR has a base of #4. When that merges, the tests should run here.

File: __tests__/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "plugins": ["jest"],
+  "env": {
+    "jest/globals": true
+  }
```
Comment (erunion) [inline __tests__/.eslintrc:5]: This'll handle loading the jest plugin + setting up globals for you:

```suggestion
  "extends": ["@readme/eslint-config/testing"],
```

File: __tests__/index.test.js [hunk 2]
```diff
@@ -1,62 +1,91 @@
...
-const action = require('../');
-
 /* We only want to test the oas property on the
  * body because there are a bunch of other
  * things that are undefined on localhost,
  * but populated when running these tests in GitHub CI
  */
 function filteringRequestBody(body) {
+  // eslint-disable-next-line no-underscore-dangle
   const _body = JSON.parse(body);
```
Comment (cvan) [inline __tests__/index.test.js:17]: could save a line by naming it parsedBody :-P

File: package.json [hunk 3]
```diff
@@ -4,6 +4,10 @@
   "description": "A GitHub action for uploading OAS files to ReadMe Micro",
   "main": "index.js",
   "scripts": {
+    "pretest": "npm run lint && npm run prettier",
```
Comment (cvan) [inline package.json:7]: this feels like it should be its own command to run in isolation of the tests. I suppose you could run 'npm run pretest'.

File: utils.js [hunk 4]
```diff
@@ -44,3 +29,27 @@ function filterOas(files) {
...
+
+    if (!globs || !globs.length) {
+      // eslint-disable-next-line no-param-reassign
+      globs = ['**/*.{yaml,yml,json}'];
+    }
+
+    let out = [];
+    globs.forEach(g => {
+      // Concat them and filter dupes
+      const list = filterOas(glob.sync(g));
```
Comment (cvan) [inline utils.js:50]: does 'glob.sync' use 'process.cwd()'? I assume it does. if it doesn't it should.

**Review by cvan — APPROVED** [review]

**Pull request merged**